Std/Data/Bool.lean

```diff
@@ -38,6 +38,50 @@ instance (x y : Bool) : Decidable (x < y) := inferInstanceAs (Decidable (!x && y
 instance : Max Bool := ⟨or⟩
 instance : Min Bool := ⟨and⟩
 
+theorem false_ne_true : false ≠ true := Bool.noConfusion
+
+theorem eq_false_or_eq_true : (b : Bool) → b = true ∨ b = false := by decide
+
+theorem eq_false_iff : {b : Bool} → b = false ↔ b ≠ true := by decide
+
+theorem eq_true_iff : {b : Bool} → b = true ↔ b ≠ false := by decide
+
+/-! ### beq -/
+
+@[simp] theorem false_beq : ∀ (x : Bool), (false == x) = !x := by decide
+
+@[simp] theorem beq_false : ∀ (x : Bool), (x == false) = !x := by decide
+
+@[simp] theorem true_beq : ∀ (x : Bool), (true == x) = x := by decide
+
+@[simp] theorem beq_true : ∀ (x : Bool), (x == true) = x := by decide
+
+theorem not_beq : ∀ (x y : Bool), ((!x) == y) = !(x == y) := by decide
+
+theorem beq_not : ∀ (x y : Bool), (x == (!y)) = !(x == y) := by decide
+
+@[simp] theorem not_beq_not : ∀ (x y : Bool), ((!x) == (!y)) = (x == y) := by decide
+
+theorem beq_eq_not_xor : ∀ (x y : Bool), (x == y) = !xor x y := by decide
```

Comment:
```suggestion
@[simp] theorem not_xor_eq_beq : ∀ (x y : Bool), !(xor x y) = (x == y)  := by decide
```

It seems like flipping the orientation and adding simp will strictly reduce term size.